LEAF_Request_Portal/templates/print_form.tpl

```diff
@@ -461,42 +484,296 @@ function cancelRequest() {
+        });
+    }
+
+    dialog.setTitle('Duplicate From <!--{$title|escape:'quotes'}-->');
+    dialog.setContent('Select new service: <br /><div id="changeService"></div>');
+    dialog.setContent(''
+        + 'Title:<br />'
+        + '<input id="title" name="title" type="text" value="<!--{$title|escape:'quotes'}-->" /><br /><br />'
+        + '<div id="serviceWrapper">Service:<br />'
+        + '<select class="chosen" id="service" name="service">'+serviceOptions+'</select><br /><br /></div>'
+        + 'Priority:<br />'
+        + '<select class="chosen" id="priority" name="priority"><option value="-10">EMERGENCY</option><option value="0" selected="selected">Normal</option></select><br /><br />'
+        + 'Sections to Copy:<br />'
+        + pickAndChooseOptions
+        + '<br /><br />'
+    );
+    dialog.show();
+    dialog.indicateBusy();
+    dialog.indicateIdle();
+
+    // hide service options if they are not available to choose from.
+    if(!(serviceOptions.length > 0)){
+        $('#serviceWrapper').hide();
+    }
+    $('.chosen').chosen({disable_search_threshold: 6});
+    dialog.setSaveHandler(function() {
+
+        // we will add on the categories in the first ajax call, this takes in what data the end user updates
+        let createData = {
+            title: $('#title').val(),
+            service: $('#service').val(),
+            priority: $('#priority').val(),
+            CSRFToken: '<!--{$CSRFToken}-->'
+        };
+
+        let updateData = {
+            series: series,
+            CSRFToken: '<!--{$CSRFToken}-->'
+        };
+
+        let fileData = [];
+        let chosenSections = [];
+        let pickAndChooseValues = $("input[name='pickAndChoose[]']:checked")
+            .map(function(){
+                return chosenSections.concat(JSON.parse($(this).val()));
+            }).get();
+
+        // get our data for submission
+        // I can probably use this to also allow for pick and choose
+        if(pickAndChooseValues.length > 0) {
+            $.ajax({
+                type: 'GET',
+                url: 'api/form/<!--{$recordID|strip_tags}-->/data',
+                CSRFToken: '<!--{$CSRFToken}-->',
+                async: false, // I am not going to nest these to make things easier to follow.
+                success: function (res) {
+                    Object.values(res).forEach(function (resultValue) {
+
+                        if (pickAndChooseValues.includes(resultValue[series].indicatorID)) {
+
+                            // uploaded files will need to have a special case done to them to copy them over to the new record
+                            if(resultValue[series].format == 'fileupload' || resultValue[series].format == 'image'){
+                                if(resultValue[series].value.length > 0){
+                                    resultValue[series].value.forEach(function(currentFile){
+                                        var fileDat = {
```

Comment:
Why is this a var instead of let?  Looks like it is only used here.

Reply:
This was something I missed, was testing something and after I realized what I was doing I forgot to change it back.